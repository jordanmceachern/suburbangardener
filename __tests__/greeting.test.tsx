import { render } from "@testing-library/react";

// Simple test component for testing the setup
function TestComponent() {
  return (
    <div className="flex items-center justify-center p-4 bg-green-100">
      <h1 className="text-xl font-bold text-green-800">Test Component</h1>
    </div>
  );
}

describe("Test Setup", () => {
  it("renders test component with correct structure and classes", () => {
    const { container } = render(<TestComponent />);

    // Take a snapshot of the DOM structure including classes
    expect(container.firstChild).toMatchSnapshot();
  });

  it("contains expected Tailwind classes", () => {
    const { container } = render(<TestComponent />);

    // Check for Tailwind classes
    const hasFlexClasses = container.innerHTML.includes("flex");
    const hasColorClasses = container.innerHTML.includes("green-");
    const hasSpacingClasses = container.innerHTML.includes("p-");

    expect(hasFlexClasses).toBe(true);
    expect(hasColorClasses).toBe(true);
    expect(hasSpacingClasses).toBe(true);
  });
});
