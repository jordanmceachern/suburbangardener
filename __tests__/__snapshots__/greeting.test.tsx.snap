// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test Setup renders test component with correct structure and classes 1`] = `
<div
  class="flex items-center justify-center p-4 bg-green-100"
>
  <h1
    class="text-xl font-bold text-green-800"
  >
    Test Component
  </h1>
</div>
`;
